### DISCUSSION TITLE
Next: Scrum?

### DISCUSSION THREAD

**Discussion by robmoffat** [risk-first/website#68] [Team Posts]
I'm not sure exactly how people are finding Risk-First on the internet.  Seems like I get a lot of traffic referred from Github.   For some reason, even fairly specific searches don't make much of an impact in Google.  

But nevertheless, I'm still seeing a fair bit of traffic over the last couple of weeks, and despite the content being free, people are buying the book.  

It's got me thinking about what I want to do next on this project.  A quick recap:

- Part one of Risk-First was explaining how Software Development is really an exercise in risk management.  Although that sounds a bit dull (maybe complex even), I try to explain it really simply.  

- Part two looks at the types of risks we face in Software Development.  I spend a while breaking down the different types.

So next, Part 3 should look at the different _techniques/practices/actions_ we use, and explain the qualities of them.  Generally, all activity boils down into something like this:

![Taking Action](https://riskfirst.org/images/generated/introduction/all_risk_management_language.png)

And Part 4 was about examining software _methodologies_ (e.g. DevOps, XP, etc) as collections of techniques.  (Some people don't like this use of the word "methodology".  I can see why, but, it's the most commonly used term for this type of thing).

Put together it looks like:

![Basic Model of Software Risks and Practices](https://riskfirst.org/images/generated/executive-summary/pattern_language.png)

I think I'm probably going to start tackling these two parts (3&4) simultaneously.  I'm going to write an article on Scrum first, looking at some of the key practices it espouses such as:

- Timeboxing
- Backlogs / Backlog Grooming
- Daily Standup
- Burndown Chart
- Estimates - Point/Relative
- Planning Poker

There are overlaps with other things... XP or Lean for example, but I think I've got to start somewhere.  

In the future, I'd like to cover DevOps, Prince2, SAFe and so on.  I think I may have people who can help me with a couple of these, as I'm not expert on everything.

Let me know what you think - especially if you have any good war stories or points to make on the areas above.

**Comment by ncreuschling** [comment]
I would like to see _feedback loops_ in general and all sorts of tests specifically.

**Comment by ghost** [comment]
you should also take a look at the SEMAT kernel, I did a small walkthrough here: https://neverfriday.com/2014/01/22/semat-kernel-example-walking-through-a-small-task/

it's another software methodology, not very popular but Ivar Jacobson is working on it and it seems more like a meta-methodology: https://semat.org/

**Comment by robmoffat** [comment]
Well, I'd not heard of that one before.   It seems like SEMAT/Essence is about enumerating the activities, deliverables and people required to build a software project _regardless of how you do it_.  From this, you can have checklists for those activities...  At least, that was my take-away?  

I found this video pretty helpful:  https://www.youtube.com/watch?v=Dl8-TDJoBc0

**Comment by ghost** [comment]
@robmoffat that's basically it. For example, the activities you mentioned above would fall under the "prepare to do the work", "coordinate activity" and "track progress" activity spaces under the Endeavor area of concern. However, some of those activities would also fall under the Customer and Solution areas of concern.

As I noted, it's kind of a meta-methodology.

**Comment by robmoffat** [comment]
Thanks for putting me on to that, Rudolf.  I think I probably need to download some of the flashcards to get a better idea of this.  It's certainly an interesting take on things.

cheers, have a good weekend.
